### PR TITLE
Sw 467 select disabled filter from smart suggest

### PR DESCRIPTION
**Pull request by zaifastafa**

### CODE REVIEW

File: src/Findologic/Resource/ServiceConfigResource.php [hunk 1]
```diff
@@ -64,15 +64,23 @@ public function isStaging(?string $shopkey): bool
...
+
     /**
      * @throws InvalidArgumentException
      */
     private function getFromCache(): ?ServiceConfig
     {
-        /** @var ServiceConfig $config */
         $config = $this->cache->getItem(self::CACHE_KEY)->get();
         if ($config !== null) {
-            return unserialize($config, [ServiceConfig::class]);
```
Comment (TheKeymaster) [inline src/Findologic/Resource/ServiceConfigResource.php:75]: Why was this changed and why was the docblock removed?

File: src/Findologic/Request/Handler/SearchNavigationRequestHandler.php [hunk 2]
```diff
@@ -19,6 +19,7 @@
 use FINDOLOGIC\FinSearch\Findologic\Response\ResponseParser;
 use FINDOLOGIC\FinSearch\Struct\Config;
 use FINDOLOGIC\FinSearch\Struct\QueryInfoMessage\QueryInfoMessage;
+use FINDOLOGIC\FinSearch\Utils\Utils;
```
Comment (TheKeymaster) [inline src/Findologic/Request/Handler/SearchNavigationRequestHandler.php:22]: Is this actually used?

File: src/Findologic/Api/ServiceConfig.php [hunk 3]
```diff
@@ -20,6 +20,9 @@ class ServiceConfig extends Struct
     /** @var DateTime */
     protected $expireDateTime;
 
+    /** @var array */
```
Comment (TheKeymaster) [inline src/Findologic/Api/ServiceConfig.php:23]: be more strict about the typehint. E.g.

```php
/**
 * @var array<string, string>
 */
protected $blocks;
```

File: src/FinSearch.php [hunk 4]
```diff
@@ -50,7 +50,7 @@ public function uninstall(UninstallContext $uninstallContext): void
         parent::uninstall($uninstallContext);
     }
 
-    private function getServiceXml(): string
+    private function getServiceXml($container): string
```
Comment (TheKeymaster) [inline src/FinSearch.php:53]: 1. This is not used.
1. It does not have a type hint.

File: src/Findologic/Response/Xml21ResponseParser.php [hunk 5]
```diff
@@ -146,7 +149,7 @@ private function buildCategoryQueryInfoMessage(array $params): CategoryInfoMessa
         $categoryInfoMessage = QueryInfoMessage::buildInstance(
             QueryInfoMessage::TYPE_CATEGORY,
             null,
-            $filters['cat']->getDisplay(),
+            isset($filters['cat']) ? $filters['cat']->getDisplay() : 'Category',
```
Comment (TheKeymaster) [inline src/Findologic/Response/Xml21ResponseParser.php:152]: The default should come from the Smart Suggest blocks. Have a look at how I've implemented it for Shopware 5 with Findologic-API:

https://github.com/findologic/plugin-shopware-5/blob/5cd716d7029c2e2d9c53224a9f1c4bf0381ade5e/FinSearchUnified/Bundle/SearchBundleFindologic/ResponseParser/Xml21ResponseParser.php#L203.

File: src/Findologic/Response/Xml21ResponseParser.php [hunk 6]
```diff
@@ -184,4 +187,49 @@ private function isFilterSet(array $params, string $name): bool
...
+            if (!$hasCategoryFilter && array_key_exists('cat', $flBlocks) && $this->isFilterSet($params, 'cat')) {
+                $display = $flBlocks['cat'];
+                $value = $params['cat'];
+
+                $customFilter = new CategoryFilter('cat', $display);
+                $filterValue = new FilterValue($value, $value, 'cat');
+                $customFilter->addValue($filterValue);
+                $customFilter->setHidden(true);
+
+                $flFilters->addFilter($customFilter);
```
Comment (TheKeymaster) [inline src/Findologic/Response/Xml21ResponseParser.php:217]: Create a method like `buildHiddenFilter`, which returns a `Filter`. Use this method to build the hidden filter for cat/vendor.

File: src/Utils/Utils.php [hunk 7]
```diff
@@ -9,6 +9,7 @@
 use Shopware\Core\Framework\DataAbstractionLayer\Exception\InconsistentCriteriaIdsException;
 use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
 use Shopware\Core\System\SalesChannel\SalesChannelContext;
+use Symfony\Component\DependencyInjection\ContainerInterface;
```
Comment (TheKeymaster) [inline src/Utils/Utils.php:12]: This is not used.

File: tests/Findologic/Resource/ServiceConfigResourceTest.php [hunk 8]
```diff
@@ -101,12 +105,14 @@ public function findologicConfigDataProvider(): array
...
      * @param bool[] $directIntegration
+     * @param string[] $flBlocks
      *
      * @throws InvalidArgumentException
      */
     public function testConfigIsFetchedFromFindologic(
         array $directIntegration,
-        bool $isStagingShop
+        bool $isStagingShop,
+        array $flBlocks
```
Comment (TheKeymaster) [inline tests/Findologic/Resource/ServiceConfigResourceTest.php:115]: `flBlocks` => `smartSuggestBlocks`

File: tests/Findologic/Response/Xml21ResponseParserTest.php [hunk 9]
```diff
@@ -271,6 +270,65 @@ public function testFiltersAreReturnedAsExpected(Xml21Response $response, array
...
+                'isHidden' => false
+
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider smartSuggestBlocksProvider
+     */
+    public function testSmartSuggestBlocks(
```
Comment (TheKeymaster) [inline tests/Findologic/Response/Xml21ResponseParserTest.php:302]: `testSmartSuggestBlocks` does not really tell you what this actually tests. Try to find something more descriptive.

File: tests/Findologic/Response/Xml21ResponseParserTest.php [hunk 10]
```diff
@@ -271,6 +270,65 @@ public function testFiltersAreReturnedAsExpected(Xml21Response $response, array
...
+        string $demoResponse,
+        array $flBlocks,
```
Comment (TheKeymaster) [inline tests/Findologic/Response/Xml21ResponseParserTest.php:304]: `flBlocks` => `smartSuggestBlocks`

File: tests/Findologic/Response/Xml21ResponseParserTest.php [hunk 11]
```diff
@@ -271,6 +270,65 @@ public function testFiltersAreReturnedAsExpected(Xml21Response $response, array
...
+                'flBlocks' => [],
+                'expectedFilterName' => null,
+                'isHidden' => null
+            ],
+            'Smart suggest blocks are sent and category filter is not in response' => [
+                'demoResponse' => 'demoResponseWithoutCategoryFilter.xml',
+                'flBlocks' => ['cat' => 'Category'],
+                'expectedFilterName' => 'Category',
+                'isHidden' => true
+
```
Comment (TheKeymaster) [inline tests/Findologic/Response/Xml21ResponseParserTest.php:287]: Empty line

File: tests/Findologic/Response/Xml21ResponseParserTest.php [hunk 12]
```diff
@@ -271,6 +270,65 @@ public function testFiltersAreReturnedAsExpected(Xml21Response $response, array
...
+            ],
+            'No smart suggest blocks are sent and category filter is available in response' => [
+                'demoResponse' => 'demoResponseWithCategoryFilter.xml',
+                'flBlocks' => [],
+                'expectedFilterName' => 'Kategorie',
+                'isHidden' => false
+
```
Comment (TheKeymaster) [inline tests/Findologic/Response/Xml21ResponseParserTest.php:294]: Empty line

File: tests/Findologic/Response/Xml21ResponseParserTest.php [hunk 13]
```diff
@@ -271,6 +270,65 @@ public function testFiltersAreReturnedAsExpected(Xml21Response $response, array
         $this->assertEquals($expectedFilters, $filters);
     }
 
+    public function smartSuggestBlocksProvider()
+    {
+        return [
```
Comment (TheKeymaster) [inline tests/Findologic/Response/Xml21ResponseParserTest.php:275]: Add tests for the `vendor` filter as well.

File: src/Resources/app/storefront/dist/storefront/js/fin-search.js [hunk 14]
```diff
@@ -1 +1 @@
...Selector(document,this.options.parentFilterPanelSelector);this.listing=window.PluginManager.getPluginInstanceFromElement(e,"Listing"),this.listing.registerFilter(this),this._preventDropdownClose()}},{key:"_preventDropdownClose",value:function(){var e=r.a.querySelector(this.el,this.options.dropdownSelector,!1);e&&e.addEventListener("click",(function(e){e.stopPropagation()}))}},{key:"_validateMethods",value:function(){if("function"!=typeof this.getValues)throw new Error("[".concat(this._pluginName,'] Needs the method "getValues"\''));if("function"!=typeof this.getLabels)throw new Error("[".concat(this._pluginName,'] Needs the method "getLabels"\''));if("function"!=typeof this.reset)throw new Error("[".concat(this._pluginName,'] Needs the method "reset"\''));if("function"!=typeof this.resetAll)throw new Error("[".concat(this._pluginName,'] Needs the method "resetAll"\''))}}])&&c(n.prototype,o),i&&c(n,i),t}(o.a);h={parentFilterPanelSelector:".cms-element-product-listing-wrapper",dropdownSelector:".filter-panel-item-dropdown"},(p="options")in(f=y)?Object.defineProperty(f,p,{value:h,enumerable:!0,configurable:!0,writable:!0}):f[p]=h}},[["9TfS","runtime","vendor-node","vendor-shared"]]]);
```
Comment (TheKeymaster) [inline src/Resources/app/storefront/dist/storefront/js/fin-search.js:1]: Why was this built, but there were no changes in the JS?

File: src/Resources/public/administration/js/fin-search.js [hunk 15]
```diff
@@ -1 +1 @@
...ance:e=>"string"!=typeof e||e.length<=0,checkBoolFieldInheritance:e=>"boolean"!=typeof e,openSalesChannelUrl(){if(null!==this.selectedSalesChannelId){const e=new u;e.addFilter(u.equals("id",this.selectedSalesChannelId)),e.setLimit(1),e.addAssociation("domains"),this.salesChannelRepository.search(e,Shopware.Context.api).then(e=>{const n=e.first().domains.first();this._openStagingUrl(n)})}else this._openDefaultUrl()},_openDefaultUrl(){const e=`${window.location.origin}?findologic=on`;window.open(e,"_blank")},_openStagingUrl(e){if(e){const n=`${e.url}?findologic=on`;window.open(n,"_blank")}else this._openDefaultUrl()}},computed:{showTestButton(){return this.isActive&&this.shopkeyAvailable&&this.isValidShopkey&&this.isStagingShop},salesChannelRepository(){return this.repositoryFactory.create("sales_channel")}}});var S=t("kfPE"),k=t("KBz0");const{Module:m}=Shopware;m.register("findologic-module",{type:"plugin",name:"FinSearch",title:"findologic.header",description:"findologic.general.mainMenuDescription",color:"#f7ff0f",snippets:{"de-DE":k,"en-GB":S},routes:{index:{component:"findologic-page",path:"index",meta:{parentPath:"sw.settings.index"}}}})}},[["xZag","runtime","vendors-node"]]]);
```
Comment (TheKeymaster) [inline src/Resources/public/administration/js/fin-search.js:1]: Why was this built, but there were no changes in the JS?

**Review by TheKeymaster — CHANGES_REQUESTED** [review]

File: src/FinSearch.php (hunk 4 above)
Comment (zaifastafa) [reply src/FinSearch.php:53]: yes this was due to the version changes I was doing. It is not needed will remove it.

File: src/Findologic/Resource/ServiceConfigResource.php (hunk 1 above)
Comment (zaifastafa) [reply src/Findologic/Resource/ServiceConfigResource.php:75]: the docblock was removed as the `$config` variable should be a serialized string from the cache instead of the object, so it was incorrect. And the class name change I will revert it.

File: src/Resources/public/administration/js/fin-search.js (hunk 15 above)
Comment (zaifastafa) [reply src/Resources/public/administration/js/fin-search.js:1]: Not sure. Might be due to some changes in 6.3.

File: src/Resources/app/storefront/dist/storefront/js/fin-search.js (hunk 14 above)
Comment (zaifastafa) [reply src/Resources/app/storefront/dist/storefront/js/fin-search.js:1]: Not sure. Might be due to some changes in 6.3.

**Review by TheKeymaster — CHANGES_REQUESTED** [review]

File: tests/Findologic/Response/Xml21ResponseParserTest.php (hunk 11 above)
Comment (TheKeymaster) [reply tests/Findologic/Response/Xml21ResponseParserTest.php:309]: This is marked as resolved, but hasn't been resolved.

File: tests/Findologic/Response/Xml21ResponseParserTest.php (hunk 12 above)
Comment (TheKeymaster) [reply tests/Findologic/Response/Xml21ResponseParserTest.php:318]: This is marked as resolved, but hasn't been resolved.

File: src/Findologic/Resource/ServiceConfigResource.php (hunk 1 above)
Comment (TheKeymaster) [reply src/Findologic/Resource/ServiceConfigResource.php:75]: The class name change has not been reverted.

File: src/Findologic/Response/Xml21ResponseParser.php [hunk 16]
```diff
@@ -142,11 +146,17 @@ private function buildCategoryQueryInfoMessage(array $params): CategoryInfoMessa
         $categories = explode('_', $params['cat']);
         $category = end($categories);
 
+        if (isset($filters['cat'])) {
```
[the remark this answers is not in the document]
Comment (zaifastafa) [reply src/Findologic/Response/Xml21ResponseParser.php:149]: This is not needed here as this method is already called if the filter is set. So this will always be set when called.

**Review by TheKeymaster — APPROVED** [review]
